### DISCUSSION TITLE
What's the correct strategy for writing to multiple connected tables in one transaction?

### DISCUSSION THREAD

**Discussion by fergusmeiklejohn** [supabase/supabase#3649] [Questions] [answered]
Like most schemas, the one I'm working on has lots of FK connected tables. A write operation would need to complete on all connected tables or fail.
So I'm wondering what the best strategy is for this? I suppose we use a stored procedure .rpc()?
Then how does RLS and auth work with procedures?
And what if we validate the data on the server before writing? How do we pass the auth state to the server?

Thanks!

**Comment by bnjmnt4n** [comment, marked as answer]
As you've mentioned, using a stored procedure is typically the way to do this for now, since they are executed within a transaction. The stored procedure you use should be created with `security invoker`, since using `security definer` would lead to RLS policies not being applied.

When sending requests from your client to your server, you should ensure that the user's JWT is passed, either as a cookie or maybe a request parameter. Then, on the server, you can modify the supabase API client to use the current user's JWT like mentioned here (https://github.com/supabase/supabase/discussions/1094#discussioncomment-714633). This will allow you to emulate the user when making the request, allowing existing RLS policies to be respected.
